docs/x509.rst

```diff
@@ -964,6 +967,15 @@ Exceptions
 
         Returns the OID.
 
+.. class:: UnsupportedGeneralNameType
+
+    This is raised when a certificate contains an unsupported general name
+    type in an extension.
+
+    .. attribute:: type
+
+        :type: :term:`text` or int
```

Comment:
I think we should assume that x509.py (et all) should be able to cope with
effectively arbitrary data.

On Wed, Apr 22, 2015 at 8:23 AM, Paul Kehrer <notifications@github.com>
wrote:

> In docs/x509.rst
> <https://github.com/pyca/cryptography/pull/1859#discussion_r28866752>:
>
> > @@ -964,6 +967,15 @@ Exceptions
> >
> >          Returns the OID.
> >
> > +.. class:: UnsupportedGeneralNameType
> > +
> > +    This is raised when a certificate contains an unsupported general name
> > +    type in an extension.
> > +
> > +    .. attribute:: type
> > +
> > +        :type: :term:`text` or int
>
> The integer value is standard, yes. It's also explicitly restricted in the
> RFC to the items we know about in the dict I made. A malicious ASN.1
> payload could be constructed that didn't use a valid int though.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/1859/files#r28866752>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084